### ISSUE TITLE
Loading / plotting a 299568 point file is slow and unresponsive

### ISSUE THREAD

**Issue by crroush** [danvk/dygraphs#496]
I have tried this on two different machines.  One with significantly more ram and processing power than my laptop with comparable results. (both firefox and chrome on a ubuntu 14.04 box)
My file has entries that looks like: 2014-09-11T19:19:19.736707487,1.370000
If there is a way to upload a file I will happily provide the data.
I am using the temperature example provided in the tutorial, but I substituted my csv instead of temperature.csv.  

Is this too large of a file to expect dygraphs to handle?

**Comment by danvk** [comment]
Were you able to resolve this issue?

**Comment by crroush** [comment]
no, I was not, chrome is the only one that seems to run fast, it is
horribly slow in firefox.  I need it to work in firefox...

**Comment by breznak** [comment]
Hi @danvk ,
I'd like to update this issue. I've been hit with same/similar performance problem, you can see my issue and resolving PR: 
https://github.com/nupic-community/nupic.visualizations/issues/16
https://github.com/nupic-community/nupic.visualizations/pull/25

rangeSelect/zoom/pan in a larger (200k) graph took ~1min, making the browser stall and the graph unusable. 

To resolve the issue I did the following: 
* update to DyGraphs 1.1.1 (noticeable but not huge improvement)
* disabled `highlightSeries` in https://github.com/nupic-community/nupic.visualizations/commit/4d945e66de75ac160ee8173ca544dbf1da5ae98e - that seem to be the root of the problem. 

I'm attaching CPU performance profiles (from Chromium v45 on Ubuntu 14.04 LTS, 64b) (FF still cannot deal with the data)

[CPU-20151209T013034.cpuprofile.TXT](https://github.com/danvk/dygraphs/files/56231/CPU-20151209T013034.cpuprofile.TXT)
[rangeSelect-CPU-20151209T014111.cpuprofile.TXT](https://github.com/danvk/dygraphs/files/56232/rangeSelect-CPU-20151209T014111.cpuprofile.TXT)

Now, after the graph loads, the interaction is very swift :+1:

**Comment by danvk** [comment]
So the solution was to remove `highlightSeriesOpts`? I wonder if it's the animation that's causing the problem.

Can you say more about your data set? How many series, how many points per series?

**Comment by breznak** [comment]
yes. 

> Can you say more about your data set? How many series, how many points per series?

you can see the file the PR in `examples/CSV` , it's a few series (3) and 100k+ points.

**Comment by breznak** [comment]
Another possibility is that is was no speed issue at all, but on the very dense graph it was difficult for the mouse to select an empty spot to start `rangeSelect` :question:

**Comment by hmaeda** [comment]
I have a similar issue (http://stackoverflow.com/questions/38339593/shiny-loading-bar-for-htmlwidgets/), but am using R and shiny to create the dygraph (https://rstudio.github.io/dygraphs/)....the load time is quite slow once you get to above 10k points...was there ever a solution for fast rendering of data in a dygraph?

**Comment by crroush** [comment]
I gave up...

**Comment by adamborowski** [comment]
Hello, if you can provide aggregated data from web server (on different levels and ranges), I'd like to inform you that I am developing chart-independent library to enable smooth visual exploration of large time series data sets. It's in polishing state, still lack of documentation but integration seems to be very easy. Now I am integrating my library with Dygraphs (VisJS, HighCharts, Plotly, Flot, JQPlot already integrated).
https://github.com/adamborowski/explorejs
Working demo where HighCharts are connected to 9.000.000 points on server using the library.
Cheers,
Adam
